packages/drivers/odsp-driver/src/prefetchLatestSnapshot.ts

```diff
@@ -0,0 +1,207 @@
+                return fetchSnapshot(
+                    snapshotUrl,
+                    odspResolvedUrl,
+                    snapshotOptions,
+                    storageToken,
+                    odspLogger,
+                    persistedCache,
+                    snapshotCacheEntry,
+                );
+            });
+        event.end({ attempts });
+        return success;
+    }).catch((error) => false);
+}
+
+async function fetchSnapshot(
+    snapshotUrl: string,
+    odspResolvedUrl: IOdspResolvedUrl,
+    snapshotOptions: ISnapshotOptions,
+    storageToken: string,
+    logger: TelemetryLogger,
+    persistedCache: IPersistedCache,
+    snapshotCacheEntry: ICacheEntry,
+): Promise<boolean> {
+    let success = false;
+    const url = `${snapshotUrl}/trees/latest?ump=1`;
+    const formBoundary = uuid();
+    let postBody = `--${formBoundary}\r\n`;
+    postBody += `Authorization: Bearer ${storageToken}\r\n`;
+    postBody += `X-HTTP-Method-Override: GET\r\n`;
+    const logOptions = {};
+    Object.entries(snapshotOptions).forEach(([key, value]) => {
+        if (value !== undefined) {
+            postBody += `${key}: ${value}\r\n`;
+            logOptions[`snapshotOption_${key}`] = value;
+        }
+    });
+    if (odspResolvedUrl.sharingLinkToRedeem) {
+        postBody += `sl: ${odspResolvedUrl.sharingLinkToRedeem}\r\n`;
+    }
+    postBody += `_post: 1\r\n`;
+    postBody += `\r\n--${formBoundary}--`;
+    const headers: {[index: string]: any} = {
+        "Content-Type": `multipart/form-data;boundary=${formBoundary}`,
+    };
+
+    const controller: AbortController = new AbortController();
+    setTimeout(
+        () => controller.abort(),
+        snapshotOptions.timeout,
+    );
+
+    // This event measures only successful cases of getLatest call (no tokens, no retries).
+    return PerformanceEvent.timedExecAsync(
+        logger,
+        {
+            eventName: "PrefetchTreesLatest",
+            ...logOptions,
+        },
+        async (event) => {
+            const response: IOdspResponse<IOdspSnapshot> = await fetchAndParseAsJSONHelper<IOdspSnapshot>(
+                url,
+                {
+                    body: postBody,
+                    headers,
+                    signal: controller?.signal,
+                    method: "POST",
+                },
+            );
+            const snapshot: IOdspSnapshot = response.content;
+            const spReqDuration = response.headers.get("sprequestduration");
+
+            const { numTrees, numBlobs, encodedBlobsSize, decodedBlobsSize } = evalBlobsAndTrees(snapshot);
+
+            // There are some scenarios in ODSP where we cannot cache, trees/latest will explicitly tell us when
+            // we cannot cache using an HTTP response header.
+            const canCache = response.headers.get("disablebrowsercachingofusercontent") !== "true";
+            // There maybe no snapshot - TreesLatest would return just ops.
```

Comment:
Nit: grammar. Did you mean "may be" instead of "maybe"

Reply:
Done